pkg/scheduler/scheduler.go

```diff
@@ -293,6 +321,19 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 		preemptedWorkloads.Insert(e.preemptionTargets)
 		cq.AddUsage(usage)
 
+		// Remove usage of preemption targets from the snapshot to properly account
+		// for resources being freed during the scheduling cycle. This is essential
+		// for RefreshAssignmentsDuringSchedulingCycle to work correctly.
+		// Without this, workloadFits() will fail on assignment retry because we add the admission
+		// candidate's usage in AddUsage() without removing the preemption target.
+		// It is okay to RemoveUsage because s.cache.Snapshot reinitializes snapshot at the start of schedule() invocation
+		if features.Enabled(features.RefreshAssignmentsDuringSchedulingCycle) {
+			for _, target := range e.preemptionTargets {
+				targetCQ := snapshot.ClusterQueue(target.WorkloadInfo.ClusterQueue)
+				targetCQ.RemoveUsage(target.WorkloadInfo.Usage())
```

Comment:
Could we remove the workload altogether and obviate the need for passing excluded targets?